plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -802,7 +802,7 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
       fc::microseconds   max_trx_cpu_usage = max_trx_time_ms < 0 ? fc::microseconds::maximum() : fc::milliseconds(max_trx_time_ms);
 
       auto future = transaction_metadata::start_recover_keys(trx,
-                                                             _thread_pool.get_executor(),
+                                                             chain.get_thread_pool(),
```

Comment:
Why is using the chain `thread_pool` faster than using the producer_plugin `thread_pool`?

Reply:
I think the issue is with the way it is using the same thread pool to recover the keys and then wait for the recover of the keys.